.github/actions/restore-cache/action.yml

```diff
@@ -0,0 +1,130 @@
+name: Restore cached files
+description: Restores backend, vcpkg, and electron-builder from cache
+
+inputs:
+  backend:
+    description: Set to true to cache backend
+    required: false
+    default: 'false'
+
+outputs:
+  backend:
+    description: True if backend was restored from cache
+    value: ${{ steps.backend.outputs.cache-hit }}
+  vcpkg:
+    description: True if vcpkg was restored from cache
+    value: ${{ steps.vcpkg.outputs.cache-hit }}
+  electron-builder:
+    description: True if electron-builder was restored from cache
+    value: ${{ steps.electron-builder.outputs.cache-hit }}
+
+runs:
+  using: 'composite'
+  steps:
+    # The cache folder for electron and electron-builder is different across the three platforms.
+    # Furthermore, the ELECTRON_CACHE and ELECTRON_BUILDER_CACHE environment variables don't work.
+    # So there has to be separate steps for the different platforms to set the proper cache locations,
+    # which were determined by looking at the code here:
+    # https://github.com/develar/app-builder/blob/c87157ae504895a7e9e1329437293846b8e57beb/pkg/download/artifactDownloader.go#L195
+    - name: Set Electron cache paths (macos and linux)
+      if: matrix.os != 'windows-2019'
+      shell: bash
+      run: |
+        if [[ "$RUNNER_OS" == "Linux" ]]; then
+          echo "ELECTRON_CACHE_DIR=$HOME/.cache/electron" >> $GITHUB_ENV
+          echo "ELECTRON_BUILDER_CACHE_DIR=$HOME/.cache/electron-builder" >> $GITHUB_ENV
+        elif [[ "$RUNNER_OS" == "macOS" ]]; then
+          echo "ELECTRON_CACHE_DIR=$HOME/Library/Caches/electron" >> $GITHUB_ENV
+          echo "ELECTRON_BUILDER_CACHE_DIR=$HOME/Library/Caches/electron-builder" >> $GITHUB_ENV
+        fi
+
+    - name: Set Electron cache paths (Windows)
+      if: matrix.os == 'windows-2019'
+      shell: pwsh
+      run: |
+        echo "ELECTRON_CACHE_DIR=$Env:LOCALAPPDATA\electron\Cache" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+        echo "ELECTRON_BUILDER_CACHE_DIR=$Env:LOCALAPPDATA\electron-builder\Cache" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+
+    # Since Windows doesn't have bash, use separate steps to get the afv-native submodule
+    # revision on Windows vs. macos and linux.
+    - name: Get afv-native submodule revision (windows)
+      if: matrix.os == 'windows-2019'
+      shell: pwsh
+      run: |
+        $AFVNATIVE_REV = & git rev-parse HEAD:./backend/extern/afv-native
```

Comment:
Very annoying question but how does that work if i'm working on a branch in afv-native? See the current build step where I'm manually checking out the right branch. Consider that this is also a temporary measure since the goal is to use head in any case

Reply:
It's fine because this all gets run after that branch is checked out. Basically the submodule init happens/branch checkout happens, then cache is restored based on whatever is currently there.

Reply:
Here's an example from the current run, notice the submodule init happens, then the checkout, then the SHA is pulled:

![image](https://github.com/user-attachments/assets/3d9dafbc-d7af-4b3a-a970-1080a39341f4)

Reply:
And here's the current SHA for unicom2

![image](https://github.com/user-attachments/assets/0dd72eca-caee-482a-8dfb-83503bb3f073)